package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -242,60 +245,86 @@ def read_DATA_timestep(self, n_atoms, TS_class, TS_kwargs):
 
         unitcell = self._parse_box(header)
 
-        positions = np.zeros((n_atoms, 3),
-                             dtype=np.float32, order='F')
         try:
-            self._parse_pos(sects['Atoms'], positions)
+            positions, ordering = self._parse_pos(sects['Atoms'])
         except KeyError:
             raise IOError("Position information not found")
 
         if 'Velocities' in sects:
-            velocities = np.zeros((n_atoms, 3),
-                                  dtype=np.float32, order='F')
-            self._parse_vel(sects['Velocities'], velocities)
+            velocities = self._parse_vel(sects['Velocities'], ordering)
         else:
             velocities = None
 
         ts = TS_class.from_coordinates(positions,
                                        velocities=velocities,
                                        **TS_kwargs)
-        ts._unitcell = unitcell
+        ts.dimensions = unitcell
```

Comment:
Yes.  This is just a corner case where `unitcell` and `dimensions` are the same format, so they're interchangeable.